### ISSUE TITLE
More export formats, so online visualisers can use the data

### ISSUE THREAD

**Issue by justinclift** [sqlitebrowser/dbhub.io#124]
It would potentially be really useful having more export formats available, similar to the recently added [Redash JSON](fbc049eb02f5df50c39d7a54540e3b7cdbcf9467) format.

Specifically, export formats so online visualisation solutions can make use of uploaded data:

**Open source**
* [Plotly](https://plot.ly)
* [Superset](https://github.com/apache/incubator-superset)
* [Metabase](https://metabase.com)

**Closed source**
* [ChartIO](https://chartio.com/product/visual-sql)
* [Hevo](https://hevodata.com)
* [Periscope Data](https://www.periscopedata.com)
* [Looker](https://looker.com)
* [Numeracy](https://numeracy.co)

**Comment by chrisjlocke** [comment]
Is the 'redash json' link valid?  Doesn't go anywhere for me ("No results matched your search.")
Would these be baked into DB4S, or available via extensions?  I don't know how DB4S sees extensions, or whether thats purely for SQLite extensions.

**Comment by justinclift** [comment]
Oops, fixed. :smile:

**Comment by justinclift** [comment]
> Would these be baked into DB4S, or available via extensions?

In this case, it's "baked into DBHub.io", so when someone uploads a database they're able to use that data in a Redash server.

For example, if you go here:

* https://dbhub.io/nicva/Marine%20Litter%20Survey%20(Keep%20Northern%20Ireland%20Beautiful).sqlite

Then click on the green "Download database" drop down on the right.  One of the options is now "Selected table as Redash JSON".

Anyone using Redash - either the hosted service version or on their own infrastructure - is able to pull the data over the network, and generate visualisations from it.  eg stuff like this:

![redash_example1](https://user-images.githubusercontent.com/406299/52550381-080cb500-2e2c-11e9-84e5-92295725a6b9.png)

And yep, I expect I'll be writing up a significant amount of docs and how-to guides in the near future. :smile:

Also need to figure out some approach so people can pull their private databases across the network to their own Redash server, as at the moment the permission checks in our server block access to non-public stuff.

Initial thought there is to perhaps provide an API key in people's settings page, which they can provide along with the request for the database.  The server should then allow access.  But, haven't yet really thought that through nor tried any implementation of it, so not 100% sure yet.

**Comment by chrisjlocke** [comment]
Thanks for the extra information.  When I first read the issue, I didn't spot the 'dbhub' part, so thought it was a DB4S requirement.  Never heard of Redash.  

With regard to the API key, etc, can't the certificate used in DB4S grant access?  So DB4S has given them a certificate, they've uploaded their data, so something,  something, magic fairy dust, pixies, Stargate magnets and voila - they have access.  I kind of lost it on the end there...

**Comment by justinclift** [comment]
The certificate can be useful for some stuff (eg DB4S, maybe the cli too), but Redash doesn't (yet) have a way to upload a certificate to the server.  API keys are generally the way it's done, though I've never needed to implement them before.  Will need to read up on them, etc. :smile:

**Comment by karim** [comment]
> Anyone using Redash - either the hosted service version or on their own infrastructure - is able to pull the data over the network, and generate visualisations from it.

I like it! I find visualizations of data to be very important, giving you a better look on the data.

I like the screenshot you posted and I had to ask you something, since I'm not familiar with Redash, or JavaScript, or have any idea how this was generated. How hard is it to just show the visualization directly in DBHub?

Like select a table and click on a button .. Click Me! > Visualize Data ? Or something like Plot that we have in DB4S?

Is it some sort of a JavaScript library that you feed it a file for example, then it render it locally, or a server is required?

**Comment by justinclift** [comment]
Yeah, that was the first approach I went with too.  eg try and add some kind of javascript charting thing to the DBHub.io website, and have a tab in the database display area that lets people build + display stuff there.

The problem really comes down to my just not really grokking javascript - which I consider a shitty, unsafe language and tend to avoid.  So, that kind of failed. :wink:

That was a fair while ago anyway, and while the concept is valid... it puts us in competition with those other (above mentioned) visualisation platforms, when we're manpower limited.  We're probably better off concentrating on the bits we're good at, and partner with them so they can use us as a data source instead.  That way their sales/marketing teams can include us in their promotion(s), etc, and no have to worry about us being a competitor.

**Comment by karim** [comment]
> The problem really comes down to my just not really grokking javascript

Same. :smiley: 

> while the concept is valid... it puts us in competition with those other (above mentioned) visualisation platforms, when we're manpower limited.

That's true. I understand it better now, thanks.
